src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantRecalibrator.java

```diff
@@ -0,0 +1,852 @@
+ *   --rscriptFile output.plots.R
+ * </pre>
+ *
+ * <h3>Allele-specfic usage</h3>
+ * <pre>
+ * ./gatk-launch JAVA_OPTS=-Xmx4g \
+ *   VariantRecalibrator \
+ *   --variant raw_variants.withASannotations.vcf \
+ *   -AS \
+ *   --resource hapmap,known=false,training=true,truth=true,prior=15.0:hapmap_3.3.b37.sites.vcf \
+ *   --resource omni,known=false,training=true,truth=false,prior=12.0:1000G_omni2.5.b37.sites.vcf \
+ *   --resource 1000G,known=false,training=true,truth=false,prior=10.0:1000G_phase1.snps.high_confidence.vcf
+ *   --resource dbsnp,known=true,training=false,truth=false,prior=2.0 dbsnp_135.b37.vcf \
+ *   -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR -an InbreedingCoeff \
+ *   -mode SNP \
+ *   --recalFile output.AS.recal \
+ *   -tranchesFile output.AS.tranches \
+ *   --rscriptFile output.plots.AS.R
+ * </pre>
+ * The input VCF must have been produced using allele-specific annotations in HaplotypeCaller.
+ * Note that each allele will have a separate line in the output .recal file with its own VQSLOD and culprit that will be
+ * transferred to the final VCF in ApplyRecalibration.
+ *
+ * <h3>Caveats</h3>
+ *
+ * <ul>
+ * <li>SNPs and indels must be recalibrated in separate runs (but it is not necessary to separate them into different
+ * files). Mixed records are treated as indels.</li>
+ * <li>The values used in the example above are only meant to show how the command lines are composed.
+ * They are not meant to be taken as specific recommendations of values to use in your own work, and they may be
+ * different from the values cited elsewhere in our documentation. For the latest and greatest recommendations on
+ * how to set parameter values for you own analyses, please read the Best Practices section of the documentation,
+ * especially the <a href='https://www.broadinstitute.org/gatk/guide/article?id=1259'>FAQ document</a> on VQSR parameters.</li>
+ * <li>Whole genomes and exomes take slightly different parameters, so make sure you adapt your commands accordingly! See
+ * the documents linked above for details.</li>
+ * <li>If you work with small datasets (e.g. targeted capture experiments or small number of exomes), you will run into
+ * problems. Read the docs linked above for advice on how to deal with those issues.</li>
+ * <li>In order to create the model reporting plots Rscript needs to be in your environment PATH (this is the scripting
+ * version of R, not the interactive version). See <a target="r-project" href="http://www.r-project.org">http://www.r-project.org</a>
+ * for more info on how to download and install R.</li>
+ * </ul>
+ *
+ */
+@CommandLineProgramProperties(
+        summary = "Build a recalibration model to score variant quality for filtering purposes",
+        oneLineSummary = "Build a recalibration model to score variant quality for filtering purposes",
+        programGroup = VariantProgramGroup.class
+)
+public class VariantRecalibrator extends VariantWalker {
```

Comment:
I've confirmed that we do in fact need to implement support for multiple variant inputs in VQSR. In practice these will be strictly non-overlapping inputs, but we probably just want to go ahead and implement a general `MultiVariantWalker`, since it will likely be needed by other tools as well. I've opened a new ticket under the alpha-3 VQSR epic for this task (does not need to be addressed in this branch, obviously).